### DISCUSSION TITLE
Extract create and update to new method

### DISCUSSION THREAD

**Discussion by nuradiyana** [filamentphp/filament#1162] [Ideas]
First I say thank you for this project, I really like this.

Imagine there are two tables, `customers` and `orders`, usually we will first create `customer` data, then we make `orders` by selecting `customer` data using `Select`.

But what if I want to create `customers` and `orders` together in one form? The current solution, I do override the `create` method. But the changes I made were only on this line

1. https://github.com/laravel-filament/filament/blob/2.x/packages/admin/src/Resources/Pages/CreateRecord.php#L59
2. https://github.com/laravel-filament/filament/blob/2.x/packages/admin/src/Resources/Pages/EditRecord.php#L79

How about extract this two line to dedicated `protected` method, furthermore, if there is a custom, maybe using a dedicated `action` or `repository` class, it is enough to just override the method.

Trims

Note: sorry for my bad english

**Comment by danharrin** [comment]
Hey, what is the custom code you're using instead? You can likely do this already using the `HasManyRepeater` / a `saveRelationshipsUsing()` callback - without any internal code changes.

**Reply by nuradiyana** [reply]
Just an action class to separate the business logic and I reuse it for the Rest API.

More or less like this

```php
public function create(bool $another = false): void
{
    // Rest of code

    $this->record = $this->createRecord($data);

    // Rest of code
}

protected function createRecord(array $data)
{
    return CreateOrderAndCustomerAction::handle($data);
}
```

Or maybe I'm missing something ?

**Reply by danharrin** [reply]
If you'd like to extract this to a method like `handleRecordCreation()` and PR - you're more than welcome.

**Reply by nuradiyana** [reply]
Great, I will do that, thank you
